Release 3.0.0 alpha.0 (#4784)
open-formulieren/open-forms: CHANGELOG.rst

Changelog
=========

> 3.0.0-alpha.0 (2024-10-24)

Reviewer: ```suggestion
3.0.0-alpha.0 (2024-10-25)
```
I think we can get this out today still!

==========================

This is an alpha release, meaning it is not finished yet or suitable for production use.

Detailed changes
----------------

**New features**

* [:backend:`4770`] Obtain submission price from variable. There is an extra option
  for setting the price of a product, by using a form variable. The variable can only
  be of numeric data type (integer, float) and can be chosen in the pricing logic of

>   the form.

Reviewer: ah, I was confused - this ID is the PR and not the ticket. Let's drop this entry in favour of line 28. Ideally a changelog entry is just a single line! If people want to learn more, they are supposed to search in the rest of the documentation ;)

> * [:backend:`4546`] Enable soft-required validation in the form designer. This gives
>   the ability to the form designer to mark a (document) field as soft required and this
>   will give a warning to the user if they do not upload anything.

Reviewer: ```suggestion
* [:backend:`4546`] Added configuration options for soft-required validation of (file upload) fields to the form designer.
```

> * [:backend:`4709`] Improved the generic form error message and added pointers on how
>   the problem can be solved. A link to outgoing request logs is also included for further
>   investigation.

Reviewer: ```suggestion
* [:backend:`4709`] Improved the error feedback if unexpected errors happening during form saving in the form designer.
```

> * [:backend:`4524`, :backend:`4675`] Improved the user experience of the form variable
>   select/dropdown. The variable dropdown now uses the react-select component, allowing
>   for searching within the options.

Reviewer: ```suggestion
* [:backend:`4524`, :backend:`4675`] Selecting a form variable is now more user friendly. Variables are logically grouped and a search box was added.
```

* [:backend:`4764`] You can now use a form variable as the source of the submission price
  to be paid.

> .. note:: The ``soft-required`` feature is not yet fully implemented.

Reviewer: I think we can drop this - it's an alpha version so by definition unfinished.

**Bugfixes**

> * [:backend:`4600`] Fixed not properly translated texts by performing a reload on language
>   change by making use of ``onLanguageChange``. When the user changes the language, by default,
>   the SDK reloads the content and changes the active language of the form.

Reviewer: ```suggestion
* [:backend:`4600`] Fixed not all the content on the page getting translated after changing the form language.
```

> * [:backend:`4659`] Fixed default value for textfield. This is converted to an empty string in
>   case the value is null.

Reviewer: ```suggestion
* [:backend:`4659`] Fixed ``null`` default values for text-based fields.
```

> * [:backend:`4733`] Fixed segmentation errors which caused the application to crash.

Reviewer: ```suggestion
* [:backend:`4733`] Fixed a segmentation fault that could occur in dev environments.
```

I haven't actually seen this anywhere in production, afaik it was only you that suffered from this?

* [:backend:`4711`] Fixed broken submission form row styling.
* [:backend:`4695`] Fixed performance regression in Objects API (legacy) validation.
* [:backend:`4628`] Fixed a crash when copying a form with a "block next step" logic
  action.
* [:backend:`4713`] Fixed pre-request hook not running for all "Haal Centraal BRP
  Personen bevragen" operations (fixes Token Exchange extension).
* [:backend:`3629`] Fixed submission bulk export crashing when the form has repeating
  groups.

* [:backend:`4528`] Fixed vague error/log out situation when logging in with OIDC.
* [:backend:`4744`] Fixed a performance regression in the logic check calls and general
  submission processing.
* [:backend:`4774`] Fixed ``textfield`` data not being converted to a string when
  numeric data is received from a prefill plugin.

**Project maintenance**

> * [:backend:`4375`] Remove environmental variable ``DISABLE_SENDING_HIDDEN_FIELDS`` for
>   Objects API. When this was set to ``True`` the hidden fields were not sent to the Objects
>   API. This had been added in order to give some time for updating the existing forms.

Reviewer: ```suggestion
* [:backend:`4375`] Removed environment variable ``DISABLE_SENDING_HIDDEN_FIELDS`` for
  Objects API.
```
this belongs in a section "Breaking changes", like here: https://open-forms.readthedocs.io/en/stable/archive/changes-v2.html#changes

> * Maintain trivy (security scanner), address trivy-action rate limit issues.

Reviewer: ```suggestion
* Updated Trivy image scanning CI pipeline.
```

> * [:backend:`3705`] Ensure timestamps are localized and using the localtime representation.

Reviewer: ```suggestion
* [:backend:`3705`] Ensure timestamps are consistently displayed in the correct timezone in the admin interface.
```
this can go to the bugfixes section

* [:backend:`4588`] Reduced code duplication in payment related code.

> * [:backend:`4721`] Updated docs/screenshots concerning prefill module and Objects API manual.

Reviewer: ```suggestion
* [:backend:`4721`] Updated the screenshots in the documentation for prefill and the Objects API manual.
```
